README.md

```diff
@@ -54,22 +54,31 @@ true
 
 create a proxy
 
-It will recursively create a proxy upon access.
+This function will create a proxy at top level and proxy nested objects as you access them, in order to keep track of which properties were accessed via get/has proxy handlers:
 
 #### Parameters
 
--   `obj` **T** 
--   `affected` **[WeakMap](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)&lt;[object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object), any>** 
+-   `obj` **T** - This is the original object that will be wrapped by a proxy.
+-   `affected` **[WeakMap](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)&lt;[object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object), any>** - WeakMap that will hold the tracking of which properties in the proxied object were accessed.
 -   `proxyCache` **[WeakMap](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)&lt;[object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Object), any>?** 
 
 #### Examples
 
 ```javascript
-import { createDeepProxy } from 'proxy-compare';
+import { createDeepProxy } from "proxy-compare";
+
+const orginal = { a: "1", c: "2", d: { e: "3" } };
+const affecteProps = new WeakMap();
+
+const proxy = createDeepProxy(orginal, affecteProps);
+
+proxy.a // Will mark as used and track its value.
+// This will update the affected WeakMap with orginal as key
+// and a Set with "a"
```

Comment:
It's really hard to look inside the affected WeakMap. For experiment, you can specify a Map instead and which is printable. We also have `affectedToPathList` for printing.

Reply:
Should we briefly  say on the example that for printing the WeakMap they should use  `affectedToPathList`  we can elaborate more on it in another PR when adding the documentation for `affectedToPathList`. What do you think?

Reply:
sounds good.